tests/std/tests/GH_000990_any_link_without_exceptions/test.compile.pass.cpp

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <any>
+
+using namespace std;
+
+int test() {
+    // GH-990 <any>: std::any doesn't link when exceptions are disabled
+    return any_cast<int>(any{});
+}
+
+int main() {} // COMPILE-ONLY

```

Comment:
How does a compile-only test validate linking? Should we not make this a do-nothing-at-runtime test to ensure that linking works?